### PR TITLE
Plane: Log navigation target location

### LINKED ISSUE #4358: Plane: Log target loc to NTUN
#### Issue details

When trying to diagnose controlled fly aways or navigational errors from DF logs it would be valuable to know where the plane was attempting to fly.
#### Platform

[  ] All
[  ] AntennaTracker
[  ] Copter
[ x ] Plane
[  ] Rover

### PR DESCRIPTION
**Pull request by WickedShell**
Closes #4358

### CODE REVIEW

**Comment by WickedShell** [comment]
@peterbarker can you confirm I got the multipliers correct on altitude? I'm unsure if the new preference is to just use the `int32_t` field and set the unit multiplier to `1e-2` or set the field as `int32_t * 100` with the unit multiplier of `1`.

Also added `.gdbinit` to gitignore cause that was bugging me here.

File: ArduPlane/Log.cpp [hunk 1]
```diff
@@ -292,7 +298,7 @@ const struct LogStructure Plane::log_structure[] = {
     { LOG_CTUN_MSG, sizeof(log_Control_Tuning),     
       "CTUN", "Qcccchhhf",    "TimeUS,NavRoll,Roll,NavPitch,Pitch,ThrOut,RdrOut,ThrDem,Aspd", "sdddd---n", "FBBBB---0" },
     { LOG_NTUN_MSG, sizeof(log_Nav_Tuning),         
-      "NTUN", "Qfcccfff",  "TimeUS,WpDist,TargBrg,NavBrg,AltErr,XT,XTi,ArspdErr", "smddmmmn", "F0BBB0B0" },
+      "NTUN", "QfcccfffLLi",  "TimeUS,WpDist,TargBrg,NavBrg,AltErr,XT,XTi,ArspdErr,Lat,Lng", "smddmmmnDUm", "F0BBB0B0GGB" },
```
Comment (magicrub) [inline ArduPlane/Log.cpp]: Aren't you missing the third altitude string?

Comment (WickedShell) [reply ArduPlane/Log.cpp]: Yes... I actually didn't add alt at all till this morning when I was thinking about bugging people to look at it. Anyways fixed (and still just under the 64 char limit).

File: ArduPlane/Log.cpp [hunk 2]
```diff
@@ -292,7 +298,7 @@ const struct LogStructure Plane::log_structure[] = {
     { LOG_CTUN_MSG, sizeof(log_Control_Tuning),     
       "CTUN", "Qcccchhhf",    "TimeUS,NavRoll,Roll,NavPitch,Pitch,ThrOut,RdrOut,ThrDem,Aspd", "sdddd---n", "FBBBB---0" },
     { LOG_NTUN_MSG, sizeof(log_Nav_Tuning),         
-      "NTUN", "Qfcccfff",  "TimeUS,WpDist,TargBrg,NavBrg,AltErr,XT,XTi,ArspdErr", "smddmmmn", "F0BBB0B0" },
+      "NTUN", "QfcccfffLLi",  "TimeUS,WpDist,TargBrg,NavBrg,AltErr,XT,XTi,ArspdErr,Lat,Lng,Alt", "smddmmmnDUm", "F0BBB0B0GGB" },
```
Comment (tridge) [inline ArduPlane/Log.cpp]: can we fit TLat, TLon and TAlt ?

Comment (WickedShell) [reply ArduPlane/Log.cpp]: Not unless we cut something else shorter. It's at 63 chars right now.

**Comment by peterbarker** [comment]
@WickedShell units look correct.  The units and mults field go together - so if you specify metres but your measurement is *actually* in cm you have to specify the multiplier.

The precise field type is irrelevant in this new scheme.  Keeping it consistent with how we *did* do things is a really good idea, of course.  Some time in the future I'd love to make the field-type simply specify the c-storage-type - squash all of the "special variants" of those into the simple C-types.  Have to wait until consumers of this data (log analysis tools) all use units/mults.

File: ArduPlane/Log.cpp (hunk 2 above)
Comment (OXINARF) [reply ArduPlane/Log.cpp]: TargBrg -> TBrg?

**Pull request merged**